scenario/SupplyScenario.ts

```diff
@@ -167,6 +204,45 @@ scenario(
   }
 );
 
+scenario(
+  'Comet#supply > repay borrow with token fees',
+  {
+    tokenBalances: {
+      albert: { $base: '==1000' }
+    },
+    cometBalances: {
+      albert: { $base: -1000 } // in units of asset, not wei
+    },
+    filter: async (ctx) => matchesDeployment(ctx, [{ deployment: 'usdt' }]),
+  },
+  async ({ comet, actors }, context, world) => {
+    // Set fees for USDT for testing
+    const USDTAdmin = await world.deploymentManager.getSigner();
+    const USDT = await world.deploymentManager.existing('USDT', await comet.baseToken(), 'goerli');
+    // 10 basis points, and max 10 USDT
+    await USDT.connect(USDTAdmin).setParams(10, 10);
+
+    const { albert } = actors;
+    const baseAssetAddress = await comet.baseToken();
+    const baseAsset = context.getAssetByAddress(baseAssetAddress);
+    const scale = (await comet.baseScale()).toBigInt();
+    const utilization = await comet.getUtilization();
+    const borrowRate = (await comet.getBorrowRate(utilization)).toBigInt();
+
+    expectApproximately(await albert.getCometBaseBalance(), -1000n * scale, getInterest(1000n * scale, borrowRate, 1n) + 1n);
+
+    // Albert repays 100 units of base borrow
+    await baseAsset.approve(albert, comet.address);
+    const txn = await albert.supplyAsset({ asset: baseAsset.address, amount: 1000n * scale });
+
+    // XXX all these timings are crazy
+    // Expect to have -1000000, due to token fee, alber only repay 999 USDT instead of 1000 USDT, thus alber still owe 1 USDT which is 1000000
+    expectApproximately(await albert.getCometBaseBalance(), -1000000n, getInterest(1000n * scale, borrowRate, 4n) + 2n);
```

Comment:
Can you use `exp(1, 6)` instead? The number of 0s is hard to read

Reply:
Changed to exp(1,6) but since it's negative so I added  `-1n *`